scripts/jinja_gen.py

```diff
@@ -31,20 +38,18 @@ def get_file_contents(filepath):
     parser.add_argument('--gst_udp_port', default=5600, help="Gstreamer UDP port for SITL")
     parser.add_argument('--video_uri', default=5600, help="Mavlink camera URI for SITL")
     parser.add_argument('--mavlink_cam_udp_port', default=14530, help="Mavlink camera UDP port for SITL")
-    parser.add_argument('--ros2-distro', default='', dest='ros2_distro', type=str,
-                    help="ROS2 distro, only required if generating the agent for usage with ROS2 nodes, by default empty")
+    parser.add_argument('--build_ros_interface', default=False, dest='build_ros_interface', type=str2bool,
+                    help="required if generating the agent for usage with ROS nodes, by default false")
     args = parser.parse_args()
     env = jinja2.Environment(loader=jinja2.FileSystemLoader(args.env_dir))
     template = env.get_template(os.path.relpath(args.filename, args.env_dir))
 
     # get ROS 2 version, if exists
-    ros2_distro = ''
-    ros_version = os.environ.get('ROS_VERSION')
-    if ros_version == '2':
-        if args.ros2_distro != '':
-            ros2_distro = args.ros2_distro
-        else:
-            ros2_distro = os.environ.get('ROS_DISTRO')
+    ros_distro = ''
+    if args.build_ros_interface:
+        ros_version = os.environ.get('ROS_VERSION')
+        if ros_version == '2':
```

Comment:
This is not how it should be done. We should have a way to separate between one version and the other. Also ROS_DISTRO is not relevant anymore because we are differentiating the code generation between ROS 1 ans ROS 2 (ex. Depth_camera use the openni plugin in ROS 1 and the camera plugin in ROS 2, and that's the differentiation that should exist).

Reply:
To clarify, can you please just check for the ROS_VERSION, set in in the jinja generator, and then use it in the depth_camera to differentiate between using the openni or the camera plugin according to the ROS version?

Reply:
@TSC21, I made following changes.
- argument build_ros_interface -> generate_ros_models (It should be changed before but I didn't change it. It was mistake.)
- remove checking ros_distro and make just check ros_version (sdf.jinja is also changed to reflect this)